lib/tool-bar-view.coffee

```diff
@@ -23,13 +21,13 @@ module.exports = class ToolBarView extends View
     nextItem
 
   removeItem: (item) ->
-    index = @items.indexOf item
-    @items.splice index, 1
-    element = atom.views.getView item
```

Comment:
I only touched code relevant to teardown for the purposes of freeing any references to `tool-bar` at disposal. `atom.views.getView` isn't needed here because the target view (i.e. `item`) is handling its own removal when its `destroy()` is called. So there is no need to get a reference to its DOM node here anymore.